src/Sarif/Core/PropertyBagHolder.cs

```diff
@@ -19,6 +19,12 @@ namespace Microsoft.CodeAnalysis.Sarif
     /// </summary>
     public class PropertyBagHolder : IPropertyBagHolder
     {
+        static PropertyBagHolder()
+        {
+            // Mitigation for Newtonsoft.Json v12 vulnerability GHSA-5crp-9r3c-p9vr
+            JsonConvert.DefaultSettings ??= () => new JsonSerializerSettings { MaxDepth = 64 };
```

Comment:
JsonReader.MaxDepth is instance data which is, as far as I know, initialized either from an internal Newtonsoft default *or* as a result of creating an alternate default settings object using JsonConvert.DefaultSettings. 

Let me go take a look at the Newtonsoft code, though, to prove that assumption or prove me wrong. :)

Reply:
if it is don' t think fix needed, just to note.

Reply:
OK, look at the following code. The default serializer acquires a default settings object from JsonConvert. During serialization, the serializer will override the JsonReader max depth if it differs from the default setting. This approach has existed for several years.

Please review yourself and confirm whether you think I'm right about things. :)

[https://github.com/JamesNK/Newtonsoft.Json/blob/94ff24f815a777269920f5a7b13f60db50c5eeb7/Src/Newtonsoft.Json/JsonSerializer.cs#L623](https://github.com/JamesNK/Newtonsoft.Json/blob/94ff24f815a777269920f5a7b13f60db50c5eeb7/Src/Newtonsoft.Json/JsonSerializer.cs#L623)

[https://github.com/JamesNK/Newtonsoft.Json/blob/94ff24f815a777269920f5a7b13f60db50c5eeb7/Src/Newtonsoft.Json/JsonSerializer.cs#L958](https://github.com/JamesNK/Newtonsoft.Json/blob/94ff24f815a777269920f5a7b13f60db50c5eeb7/Src/Newtonsoft.Json/JsonSerializer.cs#L958)

Reply:
is that JsonReader internal inside serializer
I see this was the PR
https://github.com/JamesNK/Newtonsoft.Json/pull/2462
the change both in the file JsonReader.cs and JsonSerializerSettings
```
protected JsonReader()
        {
            _currentState = State.Start;
            _dateTimeZoneHandling = DateTimeZoneHandling.RoundtripKind;
            _dateParseHandling = DateParseHandling.DateTime;
            _floatParseHandling = FloatParseHandling.Double;
            _maxDepth = 128;

            CloseInput = true;
        }
```
the usage of the JsonReader is like:
`using (JsonTextReader jsonReader = new JsonTextReader(streamReader))`
it may not related to the serializer.
(what the source code is like in V6 I have not looked)

Bear with me if I am wrong :)